core/templates/services/image-upload-helper.service.ts

```diff
@@ -16,174 +16,185 @@
+    return this.sanitizer.bypassSecurityTrustResourceUrl(
+      this.assetsBackendApiService.getThumbnailUrlForPreview(
+        entityType, entityId, encodedFilepath)) as string;
+  }
+
+  generateImageFilename(
+      height: number, width: number, extension: string): string {
+    return 'img_' +
+      this._generateDateTimeStringForFilename() +
+      '_height_' + height +
+      '_width_' + width +
+      '.' + extension;
+  }
+
+  cleanMathExpressionSvgString(svgString: string): string {
+    // We need to modify/remove unnecessary attributes added by mathjax
+    // from the svg tag.
+    let domParser = new DOMParser();
+    let doc = domParser.parseFromString(svgString, 'image/svg+xml');
+    doc.querySelectorAll('*').forEach((node) => {
+      if (node.tagName.toLowerCase() === 'svg') {
+        node.removeAttribute('xmlns:xlink');
+        node.removeAttribute('role');
+        // We are removing this attribute, because currently it is not in
+        // the white list of valid attributes.
+        node.removeAttribute('aria-hidden');
+        node.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
+      }
+      // Remove the custom data attributes added by MathJax.
+      // These custom attributes don't affect the rendering of the SVGs,
+      // and they are not present in the white list for allowed attributes.
+      for (let i = 0; i < node.attributes.length; i++) {
+        if (node.attributes[i].name.toLowerCase().startsWith('data-')) {
+          node.removeAttribute(node.attributes[i].name.toLowerCase());
         }
-      },
-
-      getInvalidSvgTagsAndAttrs: function(dataURI) {
-        // Convert base64/URLEncoded data component to raw binary data
-        // held in a string.
-        var svgString = atob(dataURI.split(',')[1]);
-        var domParser = new DOMParser();
-        var doc = domParser.parseFromString(svgString, 'image/svg+xml');
-        var invalidTags = [];
-        var invalidAttrs = [];
-        var allowedTags = Object.keys(Constants.SVG_ATTRS_WHITELIST);
-        var nodeTagName = null;
-        doc.querySelectorAll('*').forEach((node) => {
-          nodeTagName = node.tagName.toLowerCase();
-          if (allowedTags.indexOf(nodeTagName) !== -1) {
-            for (var i = 0; i < node.attributes.length; i++) {
-              if (Constants.SVG_ATTRS_WHITELIST[nodeTagName].indexOf(
-                node.attributes[i].name.toLowerCase()) === -1) {
-                invalidAttrs.push(
-                  node.tagName + ':' + node.attributes[i].name);
-              }
-            }
-          } else {
-            invalidTags.push(node.tagName);
-          }
-        });
-        return { tags: invalidTags, attrs: invalidAttrs };
-      },
-
-      getTrustedResourceUrlForThumbnailFilename: function(
-          imageFileName, entityType, entityId) {
-        var encodedFilepath = window.encodeURIComponent(imageFileName);
-        return $sce.trustAsResourceUrl(
```

Comment:
@kevintab95, @seanlip any idea why this is here in the first place? There is no appropriate unit test for this function as well.

Reply:
It doesn't currently work without it (I verified this manually) -- the image just doesn't show up if the trustAsResourceUrl() call is removed.

Having said that, I think it's worth auditing to understand why we need it in Angular 8 and what the best practice for this is, so that we can set a pattern here for doing it properly in the future.

Reply:
@seanlip I checked that if we remove the sanitized in angular then the problem for broken link fixes. The error it is currently showing is probably due to the directive file for thumnail editor isn't migrated.

sanitizer needs the [property] binding.

Reply:
Just FYI, as per offline discussion with @darksun27, we will be migrating the directives that use this function from image-uploader-helper.service.ts To use this function properly, we need angular 8 binding mechanism
`<some-html-tag  [src]="variable">....`
we can't use `<some-thml ng-src="" src="$ctrl.somevar">`